### PR TITLE
fix: add some constraints to our debounce and refresh token

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Trying to solve the infinite loop issue
- Main constraints:
- Add debounce to minimum of 200ms to avoid race conditions
- Add minimum call time to refresh call to avoid recalling within a minute (should be fine as our token is valid 15 minutes 😅)

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

**Comment by rebelchris** [comment]
Seems to be validated and working:
https://dailydotdev.slack.com/archives/C03C3JZPRU4/p1696938514735759

File: packages/shared/src/hooks/useRefreshToken.ts [hunk 1]
```diff
@@ -11,15 +11,17 @@ export function useRefreshToken(
...
+  const saveDelay =
+    differencePlusTwoMinutes <= 200 ? 200 : differencePlusTwoMinutes;
+
+  const [useRefresh] = useDebounce(refresh, saveDelay, 1000 * 60);
 
   useEffect(() => {
-    if (accessToken) {
+    if (accessToken?.token) {
       // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
       // eslint-disable-next-line react-hooks/rules-of-hooks
```
Comment (idoshamun) [inline packages/shared/src/hooks/useRefreshToken.ts:23]: We don't need this anymore right?

**Review by idoshamun — APPROVED** [review]
Let's hope it'll take RPS to the floor

**Review by omBratteng — APPROVED** [review]
Tested it locally, now only does one request for new access token 🦾

File: packages/shared/src/hooks/useRefreshToken.ts (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/hooks/useRefreshToken.ts:23]: We do, it's a hook call inside a hook call (technically not according to React rules 😅)

Comment (idoshamun) [reply packages/shared/src/hooks/useRefreshToken.ts:23]: You gotta love these rules

Comment (capJavert) [reply packages/shared/src/hooks/useRefreshToken.ts:23]: if you just rename it to something not starting with `use` we can remove it. `useRefresh` is not really a hook.

**Pull request merged**